### PR TITLE
Removed database driver implementations from Runner

### PR DESCRIPTION
**Pull request by ydemetriades**
Removed the following dependencies from FluentMigrator.Runner
- Db2
- Firebird
- MySql
- Oracle
- Postgres
- Redshift
- SqlAnywhere
- SQLite
- SqlServer
- SqlServerCe
- Hana

Removed obsolete dependencies classes and tests

When installing FluentMigrator.Runner, it downloads all database drivers implementation nuget packages and thus polluting the resulting published application with dll's that not being used.

If you would like to keep the classes and tests I am willing to create another library that contains the obsolete classes just to keep the backwards compatibility.

### CODE REVIEW

**Comment by jzabroski** [comment]
@ydemetriades Don't you think a change like this should go in the develop branch which targets 4.0?  When we release 3.2.2 in the next few days/weeks, this would break everyone who is dependent on this packaging behavior.  If you want to curtail the published dlls, you can do so by customizing your pack command.

That said, we're also talking about saving only 421Kb here.

File: samples/FluentMigrator.Example.Migrator/FluentMigrator.Example.Migrator.csproj [hunk 1]
```diff
@@ -25,5 +25,9 @@
   <ItemGroup>
     <ProjectReference Include="..\..\src\FluentMigrator.Runner\FluentMigrator.Runner.csproj" />
     <ProjectReference Include="..\FluentMigrator.Example.Migrations\FluentMigrator.Example.Migrations.csproj" />
+    <ProjectReference Include="..\..\src\FluentMigrator.Runner.SQLite\FluentMigrator.Runner.SQLite.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
```
Comment (jzabroski) [inline samples/FluentMigrator.Example.Migrator/FluentMigrator.Example.Migrator.csproj]: I'm not sure what you are trying to do here with the example project?

Comment (jzabroski) [reply samples/FluentMigrator.Example.Migrator/FluentMigrator.Example.Migrator.csproj]: To clarify, the example Migrator csproj demonstrates the ideal future end-state for FLuentMigrator that we're inching towards in v4.  We ran into various technical challenges in v3, mostly due to under-documented APIs and design decisions in .NET Core SDK:

* It wasn't clear from a project management perspective whether .netstandard was like a framework or merely an API specification
* It wasn't clear how to run old assemblies on newer runtimes (See #1139 )
* There is magic with how `.deps.json` determines what assemblies get loaded into the default AssemblyLoadContext.  See #1016 - Using an in-process runner pretty much guarantees a correct `.deps.json` file gets generated.

**Comment by ydemetriades** [comment]
Sorry for not responding earlier. So great news! Fluent Migrator version 3 is live! Great work! 
It seems that the main purpose of my Pull Request, which was to decouple the Implementations (eg. SqlServer, Db2) from the FluentMigrator.Runner was not solved. Have a look [here](https://github.com/fluentmigrator/fluentmigrator/blob/master/src/FluentMigrator.Runner/FluentMigrator.Runner.csproj#L15).

Please allow me some time to catch up with the changes and commit some more changes!
@jzabroski Leave me a comment if you need any further clarifications! 😎

**Comment by jzabroski** [comment]
> Sorry for not responding earlier. So great news! Fluent Migrator version 3 is live! Great work!
> It seems that the main purpose of my Pull Request, which was to decouple the Implementations (eg. SqlServer, Db2) from the FluentMigrator.Runner was not solved. Have a look [here](https://github.com/fluentmigrator/fluentmigrator/blob/master/src/FluentMigrator.Runner/FluentMigrator.Runner.csproj#L15).
> 
> Please allow me some time to catch up with the changes and commit some more changes!
> @jzabroski Leave me a comment if you need any further clarifications! 😎

I guess I'm not clear on what problem you want to solve for FM v3.  As I mentioned, the additional runner packaging costs only 421 Kb.  Is there a problem you're running into?  Otherwise, let's push this change for FM v4, as it's a breaking change for consumers.  The `develop` branch is where you want to push to.

**Pull request closed without merging**